### DISCUSSION TITLE
Share your custom widgets

### DISCUSSION THREAD

**Discussion by nikth0** [Jean-Tinland/simple-bar#396] [Show and tell]
Just as the name suggests. I'm interested to see how you guys create your own widgets. Let's share

**Comment by Jean-Tinland** [comment]
Hi @nikitathomas342! I converted your issue into a discussion as it is more exchange about the project than an issue.

On my end I'm simply using 2 custom widgets:

- A slack message indicator
- A switch between my personal and pro ssh keys

## Slack

### simple-bar configuration

![slack-widget-config](https://github.com/Jean-Tinland/simple-bar/assets/43068795/8f389ff0-7704-46b4-b47d-b5d1cc145fe0)

### slack.zsh

```shell
STATUS_LABEL=$(lsappinfo info -only StatusLabel "Slack")

if [[ $STATUS_LABEL =~ \"label\"=\"([^\"]*)\" ]]; then
    LABEL="${BASH_REMATCH[1]}"
    COUNT=""

    if [[ $LABEL == "" ]]; then
        COUNT="-"
    elif [[ $LABEL == "•" ]]; then
        COUNT=$LABEL
    elif [[ $LABEL =~ ^[0-9]+$ ]]; then
        COUNT="$LABEL"
    fi

    echo "$COUNT"
else
  echo "-"
fi
```

### Output

**No message/not started**
![no-message](https://github.com/Jean-Tinland/simple-bar/assets/43068795/7d2ff319-651f-44b6-8354-b621e54fc662)

**1 or more messages**
![one-or-more-message](https://github.com/Jean-Tinland/simple-bar/assets/43068795/fce76004-4f07-4152-b0cc-4d5ec22ab9b3)

**1 direct mention**
![one-direct-mention](https://github.com/Jean-Tinland/simple-bar/assets/43068795/1a878c30-831a-4b95-80c1-fc465c6eacb5)

I got this script from the Sketchy Bar github discussion section and I adapted it to suits my needs.

## SSH key switch

I'm not exposing this script here but this is a simple script that toggle the content of my `~/.ssh/config` file.

**Reply by nikth0** [reply]
That's cool, will definitely try it out
